# HG changeset patch
# User robm
# Date 1324518632 0
# Node ID cb20ed4b953add8f2443831a0552640efca53ab7
# Parent  6b46f3c7c97cb060f88b196171b95d33bff80b7c
7103725: REGRESSION - 6u29 breaks ssl connectivity using TLS_DH_anon_WITH_AES_128_CBC_SHA
Summary: resolving an issue with 7064341.
Reviewed-by: wetmore, coffeys

diff -r 6b46f3c7c97c -r cb20ed4b953a src/share/classes/sun/security/ssl/AppOutputStream.java
--- openjdk/jdk/src/share/classes/sun/security/ssl/AppOutputStream.java	Wed Nov 16 13:14:57 2011 +0000
+++ openjdk/jdk/src/share/classes/sun/security/ssl/AppOutputStream.java	Thu Dec 22 01:50:32 2011 +0000
@@ -90,7 +90,8 @@
             do {
                 int howmuch;
                 if (isFirstRecordOfThePayload && c.needToSplitPayload()) {
-                    howmuch = Math.min(0x01, r.availableDataBytes());
+                    howmuch = (len == 0) ? 0 : Math.min(
+                        0x01, r.availableDataBytes());
                 } else {
                     howmuch = Math.min(len, r.availableDataBytes());
                 }
